docs/adr/0001-daily-active-use-server-side-metrics-glean.md

```diff
@@ -0,0 +1,114 @@
+# Measuring Server-Side Daily Active Use (DAU) With Glean
+
+* Status: proposed
+* Deciders: Taddes Korris, David Durst, JR Conlin, Phillip Jenvey
+* Date: 2024-10-09
+
+Technical Story: 
+[Jira Epic Url](https://mozilla-hub.atlassian.net/browse/SYNC-4389)
+
+
+## Context and Problem Statement
+
+There is a requirement to move away from the current measurement of Sync Daily Active Use (DAU), which measures usage via FxA/Mozilla Accounts.
+The addition of Relay, device backups/migrations, through FxA means the metric will no longer an be accurate reflection of Sync usage. This is due to increased browser sign-ins with the potential of overcounting, or not counting those logging into the browser without Sync enabled. Furthermore, there is a broader organizational movement towards measuring DAU within services themselves. 
+
+The goal is to measure DAU (and subsequently WAU & MAU) by emitting metrics from syncserver-rs itself. This requires the following data:
+* User identifier (hashed_fxa_uid)
+* Timestamp
+* Platform (from UserAgent: Desktop, Fenix, iOS)
+* Collection updated
+ 
+In researching possible implementation methods, it became clear that many options did not offer us the ease and flexibility to reconcile the data after emission. This is why Glean is recommended as a clear frontrunner. This is not without some drawbacks, but they are minimal compared to other options that would, for example, require considerable data processing and querying difficulties. There is support for this implementation from the Glean team and active support in the process.
+
+## Decision Drivers
+
+1. Simplicity of implementation, not only for internal metric emission, but for processing and querying.
+2. Reduction of complexity and load on team to process and make sense of data. Ideally there is existing infrastructure to reduce the possible significant work to calculate DAU.
+3. Glean is an internal tool successfully used by many teams at Mozilla.
+4. The Glean team has extensive experience and can provide considerable support in establishing application metrics.
+5. Extensibility: ability to expand scope of metrics to other measurements and apply methodology to other services in Sync ecosystem.
+
+## Considered Options
+
+* A. Glean
+* B. StatsD and Grafana
+* C. Sql/Redash
```

Comment:
I'm not sure I understand "Sql/Redash" as an option considered: I see us ultimately using redash to calculate the counts from our new glean metrics, just as it is currently done from FxA's.

Reply:
Ah gotcha, I thought it was a different flow and we use Looker/BigQuery to get that result. Is that not the case? Again I've not used Redash enough for observing our metrics.  I'm happy to remove this

Reply:
Ah, I see where you're coming from: Looker vs Redash

There's a lot of overlap between the two, both are typically pulling from the same BigQuery sources. E.g. this [Sync DAU in looker](https://mozilla.cloud.looker.com/dashboards/1091?Submission%20Date=7%20day&Legacy%20Submission%20Date=2024%2F01%2F01%20to%202024%2F02%2F01) seems similar to the [Redash version](https://sql.telemetry.mozilla.org/queries/101007/source?p_end%20date=2024-06-26&p_start%20date=2024-05-01#248905), they're both reading from the same `account_events` data. I almost think of them as different UIs for I believe largely the same data sources.

(I've even seen it [suggested](https://mozilla-hub.atlassian.net/wiki/spaces/SRE/pages/27922678/STMO+-+Redash+sql.telemetry.mozilla.org#STMO-Redash(sql.telemetry.mozilla.org)-Description) that eventually Redash usage will all move to Looker but I'm not sure that will ever really happen)

I'm fine with building the new dashboard in Looker if that's the plan (or either) but I'd argue the cons section of Sql/Redash here probably don't apply? Our new glean events seem fairly similar to the one's we're relying on from fxa, so I imagine implementing the dashboard in Redash wouldn't be too difficult?